graphql_compiler/query_planning/typedefs.py

```diff
@@ -0,0 +1,310 @@
+        return None
+    dictionary_value = {}
+    for input_name, input_type in input_metadata_dictionary.items():
+        dictionary_value[input_name] = str(input_type)
+    return dictionary_value
+
+
+def _deserialize_input_metadata_field(
+    dict_value: Optional[Dict[str, str]]
+) -> Optional[Dict[str, GraphQLType]]:
+    """Deserialize input metadata, converting strings to GraphQLTypes."""
+    # It is possible to have an empty input metadata dictionary (i.e. no inputs for the query).
+    # Note that this is different than "None", which means no metadata was provided.
+    if dict_value == {}:
+        return {}
+    if dict_value is None:
+        return None
+    input_metadata_dictionary = {}
+    for input_name, input_type in dict_value.items():
+        input_metadata_dictionary[input_name] = _get_type_from_scalar_type_dictionary(
+            ALL_SCALAR_TYPES, parse_type(input_type)
+        )
+    return input_metadata_dictionary
+
+
+def _compare_input_metadata_field(
+    left_input_metadata: Optional[Dict[str, QueryArgumentGraphQLType]],
+    right_input_metadata: Optional[Dict[str, QueryArgumentGraphQLType]],
+) -> bool:
+    """Check input_metadata SimpleExecute field equality, comparing GraphQLTypes appropriately."""
+    if left_input_metadata is None:
+        # Since left_input_metadata is None, checking for equality requires determining whether or
+        # not right_metadata is also None. If right_metadata is None, left and right metadata
+        # are equal and True is returned.
+        return right_input_metadata is None
+
+    # right_input_metadata is None, but left_input_metadata is not.
+    if right_input_metadata is None:
+        return False
+
+    # Neither left_input_metadata nor right_input_metadata is None.
+    input_metadata_keys = left_input_metadata.keys()
+
+    # Check if input_metadata keys match.
+    if input_metadata_keys != right_input_metadata.keys():
+        return False
+    # Check if input_metadata values match for all keys.
+    for key in input_metadata_keys:
+        if not is_same_type(left_input_metadata[key], right_input_metadata[key]):
+            return False
+
+    # All keys and values match so return True.
+    return True
+
+
+def _deserialize_independent_query_plan_field(dict_value: Dict[str, Any]) -> "IndependentQueryPlan":
+    """Deserialize the dict representation of IndependentQueryPlan."""
+    # Note: there will be more types of IndependentQueryPlans that will require different
+    # deserialization shortly.
+    return SimpleExecute.from_dict(dict_value)
+
+
+# ############
+# Public API #
+# ############
+
+
+@unique
+class BackendType(Enum):
+    # N.B.: The values of the enums are the "human-friendly" display names. Since they are shown
+    #       to humans, they are subject to change if said humans find a friendlier name.
+    #       Don't assume they are immutable!
+    cypher = "Cypher"
+    gremlin = "Gremlin"
+    interpreter = "InterpreterAdapter"
```

Comment:
I feel like the `InterpreterAdapter` is a bit too much of an implementation detail to highlight here. The backend is an interpreter, so maybe let's just use that?
```suggestion
    interpreter = "interpreter"
```